src/sonic-yang-mgmt/sonic_yang_ext.py

```diff
@@ -378,10 +378,17 @@ def _createLeafDict(self, model, table):
         #choices, this is tricky, since leafs are under cases in tree.
         choices = model.get('choice')
         if choices:
-            for choice in choices:
-                cases = choice['case']
+            # If single choice exists in container/list
+            if isinstance(choices, dict):
+                cases = choices['case']
                 for case in cases:
                     self._fillLeafDict(case.get('leaf'), leafDict)
```

Comment:
The 2 lines are duplicated, you may consider reusing.

Reply:
You are right, but throughout the code every time, there are separation between single element (dict) and multiple (list), they are handled separately.

Reply:
no easy way to reuse.